src/store/level-wrapper.ts

```diff
@@ -170,6 +172,42 @@ export class LevelWrapper<V> {
     return abortOr(options?.signal, this.db.batch(operations));
   }
 
+  private async compactUnderlyingStorage(options?: LevelWrapperOptions): Promise<void> {
+    options?.signal?.throwIfAborted();
+
+    await abortOr(options?.signal, this.createLevelDatabase());
+
+    const range = this.sublevelRange;
+    if (!range) {
+      return;
+    }
+
+    // additional methods are only available on the root API instance
+    const root = this.root;
+
+    if (root.db.supports.additionalMethods.compactRange) {
+      return abortOr(options?.signal, root.db['compactRange']?.(...range));
+    }
```

Comment:
For my understanding, would the below not be sufficient (without the `if` check)? Are you trying to draw attention to the fact that `compactRange()` may not be available thinking that the `?.` is not explicit enough? No push back if that's the case, just trying to understand the intent.
```suggestion
    return abortOr(options?.signal, root.db['compactRange']?.(...range));
```

Reply:
i was trying to follow the API design of `level` in that you should check for `supports` before doing the thing :)